fix(css): update animation tips page (#34827)
mdn/content: files/en-us/web/css/css_animations/tips/index.md

## Run an animation again

> The [CSS Animations](/en-US/docs/Web/CSS/CSS_animations) specification doesn't offer a way to run an animation again. You can't just set the element's {{cssxref("animation-play-state")}} to `"running"` again. Instead, you have to use JavaScript to get a stopped animation to replay.

Reviewer: ```suggestion
The [CSS Animations](/en-US/docs/Web/CSS/CSS_animations) specification doesn't offer a way to run an animation again. You can't just set the element's {{cssxref("animation-play-state")}} to `"running"` again. Instead, you have to use JavaScript to get a completed animation to replay.
```
"paused" and "stopped" are synonyms, so you can get a stopped animation to continue, but you can't get a completed one to restart, unless you change the class name or animation name, and likely other JS updating dom ways.

```html
<div class="box"></div>
<button class="runButton">Run the animation</button>
```

### CSS

Let's style the box using CSS.

```css
.box {
  width: 100px;
  height: 100px;
  border: 1px solid black;
  margin-bottom: 1rem;
}
```

### JavaScript

Next, we'll look at the JavaScript that does the work. The `playAnimation()` function is to be called when the user clicks on the run button. Instead of {{cssxref("@keyframes")}} at-rule we [define the keyframes in JavaScript](/en-US/docs/Web/API/Web_Animations_API/Keyframe_Formats).

```js
const box = document.querySelector(".box");
const button = document.querySelector(".runButton");

/*
  equivalent to the following CSS @keyframes

  @keyframes colorChange {
    0% {
      background-color: grey;
    }
    100% {
      background-color: lime;
    }
  }
*/
const colorChangeFrames = { backgroundColor: ["grey", "lime"] };

function playAnimation() {
  box.animate(colorChangeFrames, 4000);
}
```

> The `playAnimation` method simply calls {{domxref("Element.animate()")}} method on the box to play the animation. The `animate` method takes the `colorChangeFrames` keyframe object and animation duration as arguments.

Reviewer: ```suggestion
The `playAnimation` method calls {{domxref("Element.animate()")}} method on the box to play the animation. The `animate` method takes the `colorChangeFrames` keyframe object and animation duration as arguments.
```
avoid calling things "simple" because it may not be for a reader.

Of course, we also need to add an event handler to our run button so it'll actually do something:

```js
button.addEventListener("click", playAnimation);
```

### Result

{{ EmbedLiveSample("Run_an_animation_again", "100%", "160") }}

## Waiting for an animation to complete before stopping

> In the previous example if the run button is clicked before the animation is completed, then the current animation is stopped abruptly and the next animation starts from the start. If instead, you'd like the current animation to be complete before starting a new one then we need to check if any animation is still running on the element before accepting a new play request.

Reviewer: ```suggestion
In the previous example, if the run button is clicked before the animation is completed, the current animation will abruptly stop and the animation restarts from the `0%` or `from` starting keyframe. If you would like the current animation iteration to be complete before starting a new one, we can disable the `run` button while the animation is running, reenabling it based on the `animationend` event. Alternatively, if we want to enable multiple iterations of the animation, we can check to see if an animation is running on the element and increment the `animation-iteration` count for each button click while the animation is running.
```
this is more of a suggestion as to how to keep this relevant to CSS animations; if we opt to keep this content in the CSS section. it's not reflective of the rest of the content on this page; which is more animation API content..
